dicoogle/src/main/resources/webapp/js/components/management/pluginsView.js

```diff
@@ -0,0 +1,141 @@
+  },
+
+  _onPluginsChange (data) {
+    this.setState({
+      error: data.error
+    });
+
+    if (!data.success) {
+      return;
+    }
+
+    if (data.data.length !== 0) {
+      let type = data.data[0].type;
+      let plugins = this.state.plugins;
+
+      // apply this structure: plugins[query]: [...] ; plugins[storage]: [...] ; ...
+      plugins[type] = data.data;
+
+      this.setState({
+        plugins: plugins
+      });
+    }
+
+    // decrease loading plugins by one unit. When all the plugins are loaded, the status will be "done"
+    this.setState(function(previousState, currentProps) {
+      let currentlyLoading = previousState.currentlyLoading - 1;
+      let status = previousState.status;
+
+      if (currentlyLoading === 0) {
+        status = "done";
+      }
+
+      return {
+        currentlyLoading: currentlyLoading,
+        status: status
+      };
+    });
+  },
+
+  _onActionClicked(type, name, action) {
+    PluginActions.setAction(type, name, action);
+  },
+
+  render () {
+    if(this.state.status === "loading"){
+      return (<div className="loader-inner ball-pulse">
+        <div/><div/><div/>
+        </div>);
+    }
+
+    const errorMessage = this.state.error ? (
+      <div className="col-md-12">
+        <div className="alert alert-info">
+          {this.state.error}
+        </div>
+      </div>
+    ) : ("");
+
+    let collapseId = 1;
+    const ignoreFieldList = ["name", "type", "enabled"];
+    const pluginPanels = Object.keys(this.state.plugins).sort().map(type => {
+      return (
+        <div className="col-lg-3 col-md-6 col-sm-12">
+          <p>{type.charAt(0).toUpperCase() + type.slice(1)}</p>
+          {this.state.plugins[type].map(plugin => (
+            <div className="panel panel-default">
+              <div className="panel-heading">
+                <h4 className="panel-title">
+                  <a className="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion"
+                     href={"#collapse" + collapseId}>{plugin.name}</a>
+                </h4>
+              </div>
+              <div id={"collapse" + collapseId++} className="panel-collapse collapse">
+                <div className="panel-body">
+                  <button type="button"
+                          className={(plugin.enabled ? "btn btn-danger" : "btn btn-success") + " pull-right"}
+                          onClick={this._onActionClicked.bind(
+                            this,
+                            type,
+                            plugin.name,
+                            (plugin.enabled ? "disable" : "enable"))}>
+                    {plugin.enabled ? "Disable" : "Enable"}
+                  </button>
+                  {
+                    Object.keys(plugin).filter(field => ignoreFieldList.indexOf(field) < 0).map(field => (
+                      <p><b>{field}:</b> {
+                        plugin[field] !== null ? plugin[field].toString() : "undefined"
```

Comment:
Let's hide undefined fields instead, as they do not provide useful information (and some of them are just placeholders for content not yet available).